doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,369 @@
+  overhead many times. This was investigated: this can be done with Highway
+  too.
+- A second concern was whether it's possible with Highway to allow the user at
+  runtime to select or disable dispatching to certain instruction sets. This is
+  possible.
+- Use of tags in Highway's C++ implementation reduces code duplication but the
+  added templating makes C-level testing and tracing more complicated.
+
+
+The ``_simd`` unit testing module
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Rewriting the ``_simd testing`` module to use C++ was done very recently in PR
+`gh-24069`_. It depends on the main PR for the move to C++, `gh-21057`_.
+It allows one to access the C++ intrinsics with almost the same signature, but
+from Python. This is a great way not only for testing, but also for designing
+new SIMD kernels.
+
+It may be possible to add a similar testing and prototyping feature to Highway
+(which uses plain ``googletest``), however currently the NumPy way is quite a
+bit nicer.
+
+
+Math routines
+~~~~~~~~~~~~~
+
+Math or numerical routines are written at a higher level of abstraction than
+the universal intrinsics that are the main focus of this NEP. Highway has only
+a limited number of math routines, and they are not precise enough for NumPy's
+needs. So either way, NumPy's existing routines (which use universal
+intrinsics) will stay, and if we go the Highway route they'll simply have to
+use Highway primitives internally. We could still use Highway sorting routines.
+If we do accept lower-precision routines (via a user-supplied choice, i.e.
+extending ``errstate`` to allow a precision option), we could use
+Highway-native routines.
+
+There may be other libraries that have numerical routines that can be reused in
+NumPy (e.g., from SLEEF, or perhaps from JPEG XL or some other Highway-using
+libraries). There may be a small benefit here, but likely it doesn't matter too
+much.
+
+
+Supported and missing intrinsics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some specific intrinsics that NumPy needs may be missing from Highway.
+
+On the other hand, Highway has more instructions that NumPy's universal
+intrinsics, so it's possible that some future needs for NumPy kernels may
+already be met there.
+
+
+Related Work
+------------
+
+- `Google Highway`_
+- `Xsimd`_
+- OpenCV's SIMD framework (`API reference <https://docs.opencv.org/4.x/df/d91/group__core__hal__intrin.html>`__, `docs <https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options>`__)
+- `std::experimental::simd <https://en.cppreference.com/w/cpp/experimental/simd/simd>`__
+- See the Related Work section in :ref:`NEP38` for more related work (as of 2019)
+
+
+Implementation
+--------------
+
+TODO
+
+
+
+Alternatives
+------------
+
+It's probably one or the other - move our universal intrinsics to C++, or use
+Google Highway for dynamic dispatch. Other alternatives include: do nothing and
+stay with C universal intrinsics, use `Xsimd`_ as the SIMD framework (less
```

Comment:
This sentence confused me, deleting the first bit un-confused me with the high-level scope of the NEP being to move to C++ in an undecided form.

```suggestion
Use Google Highway for dynamic dispatch. Other alternatives include: do nothing and
stay with C universal intrinsics, use `Xsimd`_ as the SIMD framework (less
```